packages/indexer/indexer/main.py

```diff
@@ -62,14 +63,14 @@ async def get_live_chain_data(
                     )
 
                     if not block_processed or not tx_processed:
-                        raise Exception(
+                        raise (
                             f"block or tx not processed {block_processed=} {tx_processed=}"
                         )
 
             else:
-                raise Exception("cannot pull block data")
+                raise ProcessChainDataError("cannot pull block data")
 
-        except Exception as e:
```

Comment:
Same thing here, instead of catching ALL exceptions, we are just catching the ones that we will be throwing